From 08be201824d00a0c7e5692f5dec1aa67733c932d Mon Sep 17 00:00:00 2001
From: yan <yan-wyb@foxmail.com>
Date: Fri, 4 Dec 2020 10:39:56 +0800
Subject: [PATCH 22/43] VIM3_GPIOD_OPEN_DRAIN_hdmi_fix khadas dts have
 GPIOD_OPEN_DRAIN which cant parsed properly and cant power video system1

Signed-off-by: yan <yan-wyb@foxmail.com>
---
 drivers/gpio/gpio-uclass.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpio/gpio-uclass.c b/drivers/gpio/gpio-uclass.c
index 0c01413b..750bc0eb 100644
--- a/drivers/gpio/gpio-uclass.c
+++ b/drivers/gpio/gpio-uclass.c
@@ -574,6 +574,16 @@ int dm_gpio_set_value(const struct gpio_desc *desc, int value)
 	if (ret)
 		return ret;
 
+	if (desc->flags & GPIOD_OPEN_DRAIN) {
+		if (value)
+			gpio_get_ops(desc->dev)->direction_input(desc->dev, desc->offset);
+		else
+			gpio_get_ops(desc->dev)->direction_output(desc->dev, desc->offset,0);
+
+		return 0;
+	}
+
+
 	if (desc->flags & GPIOD_ACTIVE_LOW)
 		value = !value;
 
-- 
2.17.1

